tools/python_api/src_cpp/cached_import/py_cached_item.cpp

```diff
@@ -20,4 +22,16 @@ py::handle PythonCachedItem::operator()() {
     return object;
 }
 
+py::handle NumpyMaCachedItem::operator()() {
+    // TODO: implement compatibility with numpy 2.0.0
+    KU_ASSERT((bool)PyGILState_Check());
+    auto obj = numpy();
+    if (py::cast<std::string>(obj.attr("__version__"))[0] >= '2') {
+        throw common::NotImplementedException(
```

Comment:
Use runTimeException instead of NotImplementedException since NotImplementedException is not designed to be catched by users.